### DISCUSSION TITLE
After upgrade Tailwind version, how to define utility classes with @apply?

### DISCUSSION THREAD

**Discussion by grantspilsbury** [tailwindlabs/tailwindcss#13621] [Help] [answered]
I am trying to upgrade Tailwind v2.2.15 to v3.4.3

When running the app I get this error:
```
Syntax error: /src/components/Card/Card.module.css The `text-multiline` class does not exist. If `text-multiline` is a custom class, make sure it is defined within a `@layer` directive.

.title {
  @apply text-lg
  ^
  overflow-hidden
```

I have added the utility class `text-multiline` in index.css:
```
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer utilities {
  .text-multiline {
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
  }
}
```

And then used the class with @apply like this

Card.tsx
```
...
import styles from './Card.module.css'

const Card: React.FC<Props> = ({ title }) => {
  return (
      <div className={styles.title}>
        {title}
      </div>
  )
}

export default Card
```

Card.module.css
```
.title {
  @apply text-lg
    text-multiline
}
```

**Comment by wongjn** [comment, marked as answer]
You could consider using the classes directly instead of `@apply`ing at all, like:

```jsx
const Card: React.FC<Props> = ({ title }) => {
  return (
      <div className="text-lg text-multiline">
        {title}
      </div>
  )
}

export default Card
```
It seems like it could be [premature abstraction](https://tailwindcss.com/docs/reusing-styles#avoiding-premature-abstraction).

---

Though to answer your question more directly, it seems like the `Card.module.css` file may be processed separately from `index.css` and as such, it does not see the `text-multiline` class rule in `index.css`.

**Reply by adamwathan** [reply]
Yep, here's the documentation for this issue:

https://tailwindcss.com/docs/functions-and-directives#using-apply-with-per-component-css

We don't say it in the docs (should update it maybe) but CSS modules works the same way and has the same gotchas in this situation.
